TweetMate/src/main/java/com/ybsystem/tweetmate/listeners/TweetClickListener.java

```diff
@@ -0,0 +1,107 @@
+package com.ybsystem.tweetmate.listeners;
+
+import android.view.View;
+
+import com.ybsystem.tweetmate.application.TweetMateApp;
+import com.ybsystem.tweetmate.models.entities.twitter.TwitterHashtagEntity;
+import com.ybsystem.tweetmate.models.entities.twitter.TwitterStatus;
+import com.ybsystem.tweetmate.models.entities.twitter.TwitterURLEntity;
+import com.ybsystem.tweetmate.models.enums.ClickAction;
+import com.ybsystem.tweetmate.usecases.ClickUseCase;
+import com.ybsystem.tweetmate.usecases.StatusUseCase;
+
+import static com.ybsystem.tweetmate.resources.ResString.*;
+
+public class TweetClickListener implements View.OnClickListener, View.OnLongClickListener {
+
+    private final TwitterStatus mStatus;
+    private final TwitterStatus mSource;
+    private final ClickAction mClickAction;
+
+    public TweetClickListener(TwitterStatus status, ClickAction action) {
+        this.mStatus = status;
+        this.mSource = mStatus.isRetweet() ? mStatus.getRtStatus() : mStatus;
+        this.mClickAction = action;
+    }
+
+    @Override
+    public void onClick(View view){
+        this.click();
+    }
+
+    @Override
+    public boolean onLongClick(View view) {
+        this.click();
+        return true;
+    }
+
+    private void click() {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 16 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS3776)